grpc-gcp/src/main/java/com/google/grpc/gcp/GcpManagedChannel.java

```diff
@@ -887,7 +885,13 @@ private synchronized ChannelRef pickLeastBusyChannel(boolean forFallback) {
 
   @Override
   public synchronized String authority() {
-    return channelRefs.get(0).getChannel().authority();
+    if (channelRefs.size() > 0) {
+      return channelRefs.get(0).getChannel().authority();
+    }
+    final ManagedChannel channel = delegateChannelBuilder.build();
+    final String authority = channel.authority();
+    channel.shutdownNow();
```

Comment:
Why shut down channel?

Reply:
This is a temporary channel (never used, never connected) just to fetch the authority string from.

When we have maxConcurrentStreamsLowWatermark set to zero any new request will get a new fresh channel (if max size is not reached). This also means that if we startup a pool with the first channel prepared it won't be provided for the first request. That's why I removed the initial channel. And when we don't have any channels we have to use the channel builder to fetch authority string if it's requested (it is not available on the builder itself).

Yeah, maybe shutdown is not necessary for a channel in the IDLE state but I thought maybe an implementation can have some other cleanup routines.